src/usr/share/gpgmailer/gpgkeyverifier.py

```diff
@@ -217,63 +206,63 @@ def _calculate_recipient_info(self, loop_current_time):
+    #   GPG key expiration checks are based.
+    def _build_key_expiration_warning_message(self, fingerprint, expiration_date, expiring_soon_date):
+
+        # TODO Arbitrarily taking the first email in the list for now.
+        key_associated_email = self.recipient_keys[fingerprint]['email_list']
+
+        self.logger.trace('Building expiration message for key %s to be sent to address %s.' % (fingerprint, key_associated_email))
 
         expiration_warning_message = None
         is_expired = False
         expiring_soon = False
-        fingerprint = self.email_dicts[email]['fingerprint']
+        #fingerprint = key_fingerprint['fingerprint']
 
-        if self.email_dicts[email]['is_sender'] and self.email_dicts[email]['is_recipient']:
-            address_type = 'Sender and recipient'
-        elif self.email_dicts[email]['is_sender']:
+        if fingerprint == self.recipient_keys[fingerprint]: # self.sender_key['fingerprint']:
             address_type = 'Sender'
         else:
             address_type = 'Recipient'
 
         if not self.gpgkeyring.is_current(fingerprint, expiration_date):
             is_expired = True
             expiration_warning_message = '%s key %s (%s) has expired.' % \
-                (address_type, fingerprint, email)
+                (address_type, fingerprint, key_associated_email)
             self.logger.trace(expiration_warning_message)
 
-            if not self.email_dicts[email]['expired_email_sent']:
+            if not self.recipient_keys[fingerprint]['expired_email_sent']:
                 self.logger.warn(expiration_warning_message)
-                self.email_dicts[email]['expired_email_sent'] = True
+                self.recipient_keys[fingerprint]['expired_email_sent'] = True
 
         elif not self.gpgkeyring.is_current(fingerprint, expiring_soon_date):
             expiring_soon = True
             key_expiration_date = datetime.datetime.fromtimestamp(
                 self.gpgkeyring.get_key_expiration_date(fingerprint)).strftime('%Y-%m-%d %H:%M:%S')
             expiration_warning_message = ('%s key %s (%s) will expire on %s.' % \
-                (address_type, fingerprint, email, key_expiration_date))
+                                          (address_type, fingerprint, key_associated_email, key_expiration_date))
             self.logger.trace(expiration_warning_message)
 
-            if not self.email_dicts[email]['expiring_soon_email_sent']:
+            if not self.recipient_keys[fingerprint]['expiring_soon_email_sent']:
                 self.logger.warn(expiration_warning_message)
-                self.email_dicts[email]['expiring_soon_email_sent'] = True
+                self.recipient_keys[fingerprint]['expiring_soon_email_sent'] = True
 
         else:
-            self.logger.trace('Key %s (%s) is current.' % (fingerprint, email))
+            self.logger.trace('Key %s (%s) is current.' % (fingerprint, key_associated_email))
```

Comment:
Again, list all addresses, no addresses, or first address? I'm kinda leaning on first address (but make it clear that there might be other addresses in the log message).

Reply:
Decided above to comma-delimit the addresses associated with the key for display in the email.